spec/features/appropriate_bodies/claim_an_ect_spec.rb

```diff
@@ -52,6 +30,28 @@
     end
   end
 
+  describe "when the ECT is already claimed by another appropriate body" do
+    include_context 'fake trs api client that finds teacher with specific induction status', 'InProgress'
+
+    before do
+      other_body = FactoryBot.create(:appropriate_body)
+      teacher = FactoryBot.create(:teacher, trn: '1234567')
+      FactoryBot.create(:induction_period, :active, teacher:, appropriate_body: other_body)
```

Comment:
What are your thoughts on extracting this into a method to abstract the information away from the reader, something like:

```
before do 
  create_teacher_with_induction_period
end 

private

def create_teacher_with_induction_period
  other_body = FactoryBot.create(:appropriate_body)
  teacher = FactoryBot.create(:teacher, trn: '1234567')
  FactoryBot.create(:induction_period, :active, teacher:, appropriate_body: other_body)
  teacher
end
```

Reply:
I can improve it but I prefer using `let` and trying to float the common dependencies to the top so they're shared, and can get overridden later. There is similar code we can move out of the way but I like the code the makes each context special to be easily read in the before block. I'll push a change that does this for your consideration.

Reply:
It is lazy but effectively there is no difference here. The `before` block still creates the induction period, but when it tries to resolve the `teacher` variable for example it lazily looks up the chain, rather than having it be ready and waiting. The benefit I believe with this approach is that I can decompose dependencies and common ones rise higher in the spec, then, if need be, can get nulled or tweaked. Ultimately I've seen this pattern improve terseness and readability. I also like having RSpec use the self-documenting output wherever possible. I am trying to win people over on that one.